cmake/CompileZstd.cmake

```diff
@@ -15,9 +15,12 @@ function(compile_zstd)
     FetchContent_Populate(ZSTD)
 
     add_subdirectory(${zstd_SOURCE_DIR}/build/cmake ${zstd_BINARY_DIR})
-    target_compile_options(zstd PRIVATE -Wno-array-bounds -Wno-tautological-compare)
-    target_compile_options(libzstd_static PRIVATE -Wno-array-bounds -Wno-tautological-compare)
-    target_compile_options(zstd-frugal PRIVATE -Wno-array-bounds -Wno-tautological-compare)
+
+    if (CLANG)
```

Comment:
GCC compiler is't happy with zstd library link ordering, hence, I had to limit it to CLANG for now. 

In the original code, the CLANG flag was used while adding library to the `flow` target, any reason we don't want to follow the same semantics.

Reply:
`CompileZstd` is downloading, building and installing the zstd library, my understanding is for `gcc` it is ordering of the libraries that is causing compilation issues. Hence, `CLANG` check is better done in `cmake/flow` than in this file.

Reply:
I am unsure about this since I did the test, and with this patch gcc is able to build FDB without compile error.

Reply:
This clang specification is here because by default, clang would treat some warnings as errors, e.g. array bound access warnings. With the errors, we could not build FoundationDB using clang in our Okteto build environment. With this patch, it is possible to build FDB in Okteto build environment, using both gcc and clang.